samples/test/util.py

```diff
@@ -282,22 +284,33 @@ def run_v2_pipeline(
 ):
     import tempfile
     import subprocess
-    original_pipeline_job = tempfile.mktemp(suffix='.json',
-                                            prefix="original_pipeline_job")
-    kfp.v2.compiler.Compiler().compile(pipeline_func=fn,
-                                       package_path=original_pipeline_job)
+    original_pipeline_spec = tempfile.mktemp(
+        suffix='.json', prefix="original_pipeline_spec")
+    kfp.v2.compiler.Compiler().compile(
+        pipeline_func=fn, package_path=original_pipeline_spec)
 
     # remove following overriding logic once we use create_run_from_job_spec to trigger kfp pipeline run
-    with open(original_pipeline_job) as f:
-        pipeline_job_dict = json.load(f)
+    with open(original_pipeline_spec) as f:
+        pipeline_job_dict = {
+            'pipelineSpec': json.load(f),
+            'runtimeConfig': {},
+        }
+
     for component in [pipeline_job_dict['pipelineSpec']['root']] + list(
             pipeline_job_dict['pipelineSpec']['components'].values()):
         if 'dag' in component:
             for task in component['dag']['tasks'].values():
                 task['cachingOptions'] = {'enableCache': enable_caching}
+    for input_name, input_spec in pipeline_job_dict['pipelineSpec']['root'].get(
+            'inputDefinitions', {}).get('parameters', {}).items():
+        if 'defaultValue' in input_spec:
+            if 'parameterValues' not in pipeline_job_dict['runtimeConfig']:
+                pipeline_job_dict['runtimeConfig']['parameterValues'] = {}
+            pipeline_job_dict['runtimeConfig']['parameterValues'][
+                input_name] = input_spec['defaultValue']
     for k, v in arguments.items():
-        parameter_value = pipeline_job_dict['runtimeConfig'][
-            'parameterValues'][k]
+        parameter_value = pipeline_job_dict['runtimeConfig']['parameterValues'][
+            k]
         pipeline_job_dict['runtimeConfig']['parameterValues'][
             k] = parameter_value
```

Comment:
I think the code here is mistaken, it should be:
```suggestion
        pipeline_job_dict['runtimeConfig']['parameterValues'][
            k] = v
```